src/charm.py

```diff
@@ -89,56 +235,101 @@ def _on_postgresql_pebble_ready(self, event: WorkloadEvent) -> None:
         # Create a new config layer.
         new_layer = self._postgresql_layer()
 
-        if container.can_connect():
-            # Get the current layer.
-            current_layer = container.get_plan()
-            # Check if there are any changes to layer services.
-            if current_layer.services != new_layer.services:
-                # Changes were made, add the new layer.
-                container.add_layer(self._postgresql_service, new_layer, combine=True)
-                logging.info("Added updated layer 'postgresql' to Pebble plan")
-                # TODO: move this file generation to on config changed hook
-                # when adding configs to this charm.
-                self._patroni.render_patroni_yml_file()
-                # Restart it and report a new status to Juju.
-                container.restart(self._postgresql_service)
-                logging.info("Restarted postgresql service")
-            # All is well, set an ActiveStatus.
-            self.unit.status = ActiveStatus()
-        else:
+        # Defer the event if pebble is not available yet.
+        if not container.can_connect():
```

Comment:
Is this possible/needed? I'm assuming that you only get a `pebble_ready` event if the container can connect.

Reply:
You're right Paulo. Thanks for catching this. I removed this check on https://github.com/canonical/postgresql-k8s-operator/pull/9/commits/41426336c9ff769597c3990ddcfd931ecb498e48.